Khan/perseus
# Add PhET simulation content editor

## Summary:
Create the content editor for PhET simulations, allowing users to set:
* The URL of the simulation 
* A description of the simulation (used to communicate the widget's purpose to screen readers and Khanmigo) 

I have documented the Perseus widget creation process [here](https://khanacademy.atlassian.net/wiki/spaces/LC/pages/3075276818/Creating+a+New+Widget+Type).

Issue: [LEMS-2168](https://khanacademy.atlassian.net/browse/LEMS-2168)

<img width="1041" alt="Editor page storybook page, displaying PhET widget, which contains an iframe displaying the Projectile Data Lab PhET simulation. The URL of the widget is the URL for Projectile Data Lab, and the description is 'Projectile Data Lab.'" src="https://github.com/user-attachments/assets/f39c5dcf-fee7-4e16-8f6a-bfd682137717">

<img width="1041" alt="Editor page storybook page, displaying PhET widget, which contains an iframe saying 'Sorry, this simulation cannot load.' The URL of the widget is the Google homepage URL, and the description is 'Google.'" src="https://github.com/user-attachments/assets/4a2d7338-07e6-4a95-8476-fb02752a9a7c">

## Test plan:

- Verify that the widget content editor shows up appropriately in Storybook's EditorPage (`/docs/perseuseditor-editorpage--docs`)
  - Set URL to a valid PhET simulation (like https://phet.colorado.edu/sims/html/projectile-data-lab/latest/projectile-data-lab_all.html) and verify that the simulation appears correctly
  - Change URL to a non-PhET origin and verify that the error shows up
  - Change URL to a non-existent resource from the PhET origin and verify that the error shows up
  - Click the fullscreen button and verify that the iframe goes into fullscreen mode
- `yarn jest packages/perseus-editor/src/widgets/__tests__/phet-sim-editor.test.tsx`

[LEMS-2168]: https://khanacademy.atlassian.net/browse/LEMS-2168?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
